src/PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp

```diff
@@ -0,0 +1,194 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+// IWYU pragma:  no_include <pup.h>
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+namespace gr {
+/*!
+ * \brief Contains helper functions for transforming tensors in Kerr
+ * spacetime to Kerr-Schild coordinates.
+ *
+ * Consider the Kerr-Schild form of the Kerr metric in Cartesian
+ * coordinates \f$ (t, x, y, z)\f$,
+ *
+ * \f{align*}
+ * ds^2 = -dt^2 + dx^2 + dy^2 + dz^2 + \dfrac{2Mr^3}{r^4 + a^2 z^2}
+ * \left[dt + \dfrac{r(x\,dx + y\,dy)}{r^2 + a^2} +
+ * \dfrac{a(y\,dx - x\,dy)}{r^2 + a^2} + \dfrac{z}{r}dz\right]^2,
+ * \f}
+ *
+ * where \f$r\f$ is the function defined through
+ *
+ * \f{align*}
+ * r^2(x, y, z) = \frac{x^2 + y^2 + z^2 - a^2}{2} +
+ * \sqrt{\left(\frac{x^2 + y^2 + z^2 - a^2}{2}\right)^2 + a^2 z^2}.
+ * \f}
+ *
+ * Depending on the physical context, the following coordinate transformations
+ * are usually adopted:
+ *
+ * - \em Spherical \em coordinates, defined by the usual transformation
+ *
+ *   \f{align*}
+ *   x &= r_\text{sph}\sin\theta\cos\phi,\\
+ *   y &= r_\text{sph}\sin\theta\sin\phi,\\
+ *   z &= r_\text{sph}\cos\theta.
+ *   \f}
+ *
+ *   Note that \f$r_\text{sph} \neq r\f$, so the
+ *   horizon of a Kerr black hole is \em not at constant \f$r_\text{sph}\f$.
+ *
+ * - \em Oblate \em spheroidal \em coordinates,
+ *
+ *   \f{align*}
+ *   x &= \sqrt{r^2 + a^2}\sin\vartheta\cos\varphi,\\
+ *   y &= \sqrt{r^2 + a^2}\sin\vartheta\sin\varphi,\\
+ *   z &= r\cos\vartheta.
+ *   \f}
+ *
+ *   Notice that \f$r\f$ is that defined in the Kerr-Schild form of the Kerr
+ *   metric, so the horizon is at constant \f$r\f$. (As it will be noted below,
+ *   \f$r\f$ corresponds to the Boyer-Lindquist radial coordinate.) Also, note
+ *   that the azimuthal angle \f$\varphi\f$ is the same as the corresponding
+ *   angle in spherical coordinates, \f$\phi\f$, whereas the zenithal angle
+ *   \f$\vartheta\f$ differs from the corresponding angle \f$\theta\f$.
+ *
+ * - \em Kerr \em coordinates (sometimes also referred to as
+ *   "Kerr-Schild coordinates"), defined by
```

Comment:
Please add a sentence like "These are the coordinates used in the KerrSchildCoords class" to make it extra clear.